pkg/tasks/prometheusadapter.go

```diff
@@ -11,24 +11,34 @@ import (
 )
 
 type PrometheusAdapterTask struct {
-	client    *client.Client
-	ctx       context.Context
-	factory   *manifests.Factory
-	config    *manifests.Config
-	namespace string
+	client               *client.Client
+	metricsServerEnabled bool
```

Comment:
for consistency with the metrics-server task, I would have the flag defining that prometheus-adapter is enabled
```suggestion
	enabled bool
```

Reply:
Using this flag the p-a resource removal could also be handled here instead of the metricsserver task.

Reply:
we would still need to call the delete method from metrics server since we need to wait till metrics-server is ready to serve requests